server.js

```diff
@@ -2,34 +2,237 @@ import express from 'express'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
+
+
+const Book = mongoose.model('Book', {
+  bookID: Number,
+  title: {
+    type: String
+
+  },
+  authors: String,
+  average_rating: {
+    type: Number,
+    default: 0,
+  },
+  num_pages: {
+    type: Number,
+    default: 0
+  },
+  ratings_count: {
+    type: Number,
+    default: 0
+  },
+  text_reviews_count: {
+    type: Number,
+    default: 0
+  },
+  img_url: {
+    type: String,
+    default: ''
+  }
+})
+
+const User = mongoose.model('User', {
+  username: {
+    type: String,
+    minlength: 3,
+    maxlength: 20
+  },
+  password: {
+    type: String,
+    minlength: 8
+  },
+  ratings: {
+    type: Array,
+    default: []
+  },
+  profilePicture: {
+    type: String,
+    default: ''
+  }
+})
+
+
+const createRandomBook = async (num, titlesArray, authorArray) => {
+  await Book.deleteMany()
+  const arr = Array.apply(null, Array(num))
+
+  arr.forEach((val, index) => {
+    new Book({
+      bookID: index,
+      title: titlesArray[parseInt(titlesArray.length * Math.random())],
+      authors: authorArray[parseInt(authorArray.length * Math.random())],
+      num_pages: parseInt(Math.random() * 2000),
+      ratings_count: parseInt(Math.random() * 10000),
+      average_rating: (Math.random() * 5).toFixed(2)
+
+    }).save()
+  })
+
+
+}
+if (process.env.SEED_RANDOM) {
+  const generateRandomBooks = () => {
+    const bookTitles = []
+    const authors = []
+    data.forEach((book) => bookTitles.push(book.title))
+    data.forEach((book) => {
+      if (!authors.includes(book.authors.split('-')[0])) {
+        authors.push(book.authors.split('-')[0])
+      }
+    })
+    createRandomBook(500, bookTitles, authors)
+  }
+  generateRandomBooks()
+}
+
+// if (process.env.RESET_DB) {
+//   const seedDatabase = async () => {
+//     await Book.deleteMany()
+
+//     data.forEach((book) => {
+//       new Book(book).save()
+//     })
+//   }
+//   seedDatabase()
+// }
+
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.json({
+    homepage: '/books',
+    queryOrder: '?order=highest, lowest, longest, shortest',
+    queryKeyword: '?keyword=an author or title',
+    queryPage: '?page=1,2,3,4...',
+    individualBook: '/books/1,2,3,4...',
+    keywordResultsCanBeOrdered: 'true'
```

Comment:
Try next time to create an HTML file or use template engine like ejs: https://ejs.co to make your root route look nicer :)